core/src/main/scala/kafka/server/DelayedCreatePartitions.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.server
+
+import kafka.common.TopicAndPartition
+import org.apache.kafka.clients.admin.NewPartition
+import org.apache.kafka.common.network.ListenerName
+import org.apache.kafka.common.protocol.Errors
+import org.apache.kafka.common.requests.ApiError
+
+import scala.collection.{Map, Seq, mutable}
+
+/**
+  * A delayed create partitions operation that can be created by the admin manager and watched
+  * in the topic purgatory
+  */
+class DelayedCreatePartitions(delayMs: Long,
+                              done: Map[String, ApiError],
+                              waiting: Map[String, NewPartition],
+                              adminManager: AdminManager,
+                              listenerName: ListenerName,
+                              responseCallback: Map[String, ApiError] => Unit)
+  extends DelayedOperation(delayMs) {
+
+  // the results of topics we were waiting for
+  var waitedResults = new mutable.HashMap[String, ApiError]();
```

Comment:
Should be `val` rather than `var`? Also don't need semicolon from here are other places in the scala code.
The initialization could be `mutable.Map.empty[String, ApiError]`.